src/ucp/wireup/wireup_cm.c

```diff
@@ -559,15 +574,24 @@ static unsigned ucp_cm_server_conn_request_progress(void *arg)
 }
 
 void ucp_cm_server_conn_request_cb(uct_listener_h listener, void *arg,
-                                   const char *local_dev_name,
-                                   uct_conn_request_h conn_request,
-                                   const uct_cm_remote_data_t *remote_data)
+                                   const uct_cm_listener_conn_request_args_t
+                                   *conn_req_args)
 {
     ucp_listener_h ucp_listener = arg;
     uct_worker_cb_id_t prog_id  = UCS_CALLBACKQ_ID_NULL;
     ucp_conn_request_h ucp_conn_request;
+    uct_conn_request_h conn_request;
+    const uct_cm_remote_data_t *remote_data;
     ucs_status_t status;
 
+    ucs_assert(ucs_test_all_flags(conn_req_args->field_mask,
```

Comment:
@alinask @yosefe Any particular reason why you use an assert and not regular if with some useful error reporting ? I don't think is critical path.

Reply:
@shamisp it's because these flags are set by underlying transport and not by user, so if they are not present it's considered bug in UCX and not user error
we can use ucs_assert_always, though

Reply:
@yosefe valid point. Probably ucs_assert_always will help cache bugs a bit earlier. Not must but IMHO not bad idea.